src/Microsoft.Azure.WebJobs.Host/Listeners/ListenerFactoryContext.cs

```diff
@@ -13,21 +15,37 @@ namespace Microsoft.Azure.WebJobs.Host.Listeners
     /// </summary>
     public class ListenerFactoryContext
     {
-        /// <summary>
-        /// Creates a new instance
-        /// </summary>
-        /// <param name="descriptor">The <see cref="FunctionDescriptor"/> to create a listener for.</param>
-        /// <param name="executor">The <see cref="ITriggeredFunctionExecutor"/> that should be used to invoke the
-        /// target job function when the trigger fires.</param>
-        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use.</param>
-        public ListenerFactoryContext(FunctionDescriptor descriptor, ITriggeredFunctionExecutor executor, CancellationToken cancellationToken)
+        private readonly SharedQueueHandler _sharedQueue;
+        private DispatchProcessorFactory _dispatchProcessorFactory;
+
+        internal ListenerFactoryContext(FunctionDescriptor descriptor, ITriggeredFunctionExecutor executor, SharedQueueHandler sharedQueue, CancellationToken cancellationToken)
         {
+            if (descriptor == null)
+            {
+                throw new ArgumentNullException("descriptor");
+            }
             Descriptor = descriptor;
             Executor = executor;
+            _sharedQueue = sharedQueue;
             CancellationToken = cancellationToken;
         }
 
         /// <summary>
+        /// Shared Queue for dispatching
+        /// </summary>
+        public DispatchProcessorFactory DispatchProcessorFactory
+        {
+            get
+            {
+                if (_dispatchProcessorFactory == null)
+                {
+                    _dispatchProcessorFactory = new DispatchProcessorFactory(_sharedQueue, Descriptor.Id);
+                }
```

Comment:
CAn this be in the constructor? And then the property is trivial.

Reply:
It was in the constructor, but I realized that most users of `ListenerFactoryContext` will not use the `dispatchQueue`, so I moved it out to be more lazy